### PR TITLE
XWIKI-14138 : Allow proxying the versions metadata 2

### PR DESCRIPTION
**Pull request by krisss2121**
This should be more less end. By mistake I did stupid `git push -f origin master` which reset I bad way master and the previous PR became closed: https://github.com/xwiki/xwiki-platform/pull/568

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-repository/xwiki-platform-repository-server-api/src/main/java/org/xwiki/repository/internal/RepositoryManager.java [hunk 1]
```diff
@@ -1102,4 +1239,42 @@ private boolean updateCollection(BaseObject extensionObject, String fieldName, C
...
+        }
+
+        BaseObject extensionVersionObject =
+                extensionDocument.getObject(XWikiRepositoryModel.EXTENSIONVERSION_CLASSNAME, "version", version, false);
+
+        if (extensionVersionObject == null && allowProxying && isVersionProxyingEnabled(extensionDocument)) {
+            //no ExtensionVersionClass object so we need to create such object temporarily and delete it
+            try {
+                //FIXME - see XWIKI-14138
+                XWikiDocument extensionDocumentClone = extensionDocument.clone();
```
Comment (krisss2121) [inline xwiki-platform-core/xwiki-platform-repository/xwiki-platform-repository-server-api/src/main/java/org/xwiki/repository/internal/RepositoryManager.java]: Is it ok, that I do not delete this clone in anyway?

Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-repository/xwiki-platform-repository-server-api/src/main/java/org/xwiki/repository/internal/RepositoryManager.java]: The garbage collector will take care of it.

About the comment I would prefer a more explicit test explaining it's a hack and we need to refactor the code to something cleaner.

**Comment by tmortagne** [comment]
Note: look close to ready, waiting 8.4RC1 release before applying that or @mflorea will kill me.

I guess you executed xwiki-platform-repository-test-tests to validate your changes but would be nice to also execute xwiki-platform-extension-test-tests which also use this code a lot (that's unless you already tough about it :)).

**Comment by krisss2121** [comment]
Yes - sure I'll do that. I have been already tough about it - now I got
used to it :P

Sure - I'll run this test and revert the bad formatting which is left.

Best,
KP

2017-05-22 15:47 GMT+02:00 Thomas Mortagne <notifications@github.com>:

**Comment by krisss2121** [comment]
unfotunatelly one of the tests failed
[image: Obraz w treści 1]

I won't be able to fix it today. Tomorrow I'll get down to it from the
morning.

KP

2017-05-22 15:47 GMT+02:00 Thomas Mortagne <notifications@github.com>:

**Comment by krisss2121** [comment]
[image: Obraz w treści 1]

2017-05-22 17:36 GMT+02:00 Krzysiek Płachno <krzysiekplachno@gmail.com>:

**Comment by krisss2121** [comment]
I debuged the test using xwiki build yesterday. It turned out that the test
is not deterministic due to
https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-test/xwiki-platform-extension-test-tests/src/test/it/org/xwiki/test/ui/extension/ExtensionTest.java
line 193: RandomUtils.nextInt(20)

So today I tired to run the test again by mvn clean test - but there's
another problem, I think not related with my code:
https://pastebin.com/dqezVRZu


2017-05-22 17:36 GMT+02:00 Krzysiek Płachno <krzysiekplachno@gmail.com>:

**Comment by tmortagne** [comment]
> So today I tired to run the test again by mvn clean test - but there's another problem, I think not related with my code: https://pastebin.com/dqezVRZu 

This is definitely not related to your changes but I don't really understand why you get this.

Let's say it's OK for now. Will just need to check Jenkins after applying the pull request.

Waiting for 9.4RC1 release now.

**Comment by tmortagne** [comment]
Ha one last thing, could you make sure you add a

    @since 9.5RC1

to the javadoc of all new public methods you introduced.

**Comment by krisss2121** [comment]
Ok I'll add.
5 min and upload change.

KP

2017-05-23 13:58 GMT+02:00 Thomas Mortagne <notifications@github.com>:

**Comment by tmortagne** [comment]
Don't worry you have time, I probably won't apply this before tomorrow.

**Comment by krisss2121** [comment]
anyway - uploaded :)

2017-05-23 14:25 GMT+02:00 Thomas Mortagne <notifications@github.com>:

**Comment by tmortagne** [comment]
Looks like you also closed the pull request.

**Comment by krisss2121** [comment]
ach..... the same mistake as yesterday...
I'll fix it imediatelly

2017-05-23 14:33 GMT+02:00 Thomas Mortagne <notifications@github.com>:

**Pull request closed without merging**